intro-to-git/commits.md

```diff
@@ -0,0 +1,290 @@
+
+As a recommendation, good times to make a commit are...
+
+- After implementing the code to make one test pass
+- After writing one test and implementing the feature to make that test pass
+- Before implementing a different feature
+- After writing one or two functions
+- After beginning to make a class
+- After making function stubs, class stubs, or new files
+- Before beginning to refactor code
+- After refactoring one area of code
+
+As a recommendation, bad times to make a commit are...
+
+- In the middle of refactoring code
+- In the middle of implementing a feature, and running the project produces an error
+- After implementing multiple features (these should be smaller commits)
+
+As we keep programming, we may see that large commits can make the git history more challenging to read.
+
+## In Git, There Are Local and Staging Areas
+
+Because we get a lot of value when we intentionally construct a commit, we want to be selective about the specific code changes (lines added and removed) that go into a commit.
+
+Git allows us to be selective about changes that go into a commit. It lets us do that by creating two areas that code changes can be categorized in: the _local changes area_ and the _staging area_.
+
+The two areas give us visibility and control about what goes into a commit, before we build it.
+
+- The _local changes area_ is for new code changes that only exist on this local machine. It's the first area that code changes go to.
+- The _staging area_ is for code changes that a developer indicates should go into the next commit.
+
+Most of the code changes that happen constantly as the developer types, adds code, deletes files, etc., automatically get categorized into the _local changes area_.
+
+### Viewing Local and Staging Areas
+
+In a Git project, we can view a summary of the changes in the local and staging areas by running this in the project root:
+
+```bash
+$ git status
+```
+
+Below is an example of what the `git status` command could print out.
+
+Read this output. Then, read through the explanation below. Then re-read this output to check for understanding.
+
+```
+On branch main
+Your branch is up to date with 'origin/main'.
+
+Changes to be committed:
+  (use "git restore --staged <file>..." to unstage)
+	modified:   tests/test_wave_01.py
+
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+	modified:   tests/test_wave_05.py
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	tests/test_wave_06.py
+```
+
+This `git status` output describes the code changes:
+
+- "Changes to be committed:"
+
+  - This section describes the **staging area**
+  - These are the code changes have been moved from local to staging
+  - There's a tip from Git for the command to unstage that change, `"git restore --staged <file>..."`
```

Comment:
Good suggestion:

- I'll definitely take the `$ git ...` suggestion
- The callout material is good, I'll find a way to fit this in... or not! haha
- Just as a nice FYI for you ansel, since you're generous and are helping out with callout syntax:

Obviously, I'm a wordy person, so I love a big callout. But Learn's callout styling kinda.... makes two consecutive paragraphs in one callout box kinda smushed, moreso for my liking. like there's not really a margin/something like a linebreak between them. LOL.  Um, this (obviously) shouldn't change our content (appearance != content!!!!) buuuuuuuuuuuut I thought you'd like to know, especially since your suggestion has a bunch of those.

When I take this suggestion, if it lives in a callout, I'll likely turn the syntax explanation into a bulleted list or a table even?

Reply:
Ah.  Based on your feedback, I think I've finally noticed what a callout looks like in Learn for the first time! 😅  I can certainly understand not wanting to have callouts that end up dominating the content.  Hopefully, this can get squeezed in elsewhere!